### PR TITLE
add integration tests for route creation

### PR DESCRIPTION
**Pull request by yapei**
@spadgett   #1695 is substituted by this one, I should have addressed all comments. Could you please help review again? Thanks very much

### CODE REVIEW

File: frontend/integration-tests/views/utils.view.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+      return index;
+    }
+  }
+};
+
+export const getResourceJSON = (name: string, namespace: string, kind: string) => {
+  return execSync(`kubectl get -o json -n ${namespace} ${kind} ${name}`).toString();
+};
+
+export const isValueInJSONPath = (propertyPath: string, value: string, name: string, namespace: string, kind: string): boolean => {
```
Comment (spadgett) [inline frontend/integration-tests/views/utils.view.ts]:
```suggestion
export const isValueInJSONPath = (propertyPath: string | string[], value: string, name: string, namespace: string, resource: string): boolean => {
```

File: frontend/integration-tests/tests/networking.scenario.ts [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  });
+
+  afterAll(async() => {
+    execSync(`oc delete pod caddy-docker -n ${testName}`);
+    execSync(`oc delete service service-unsecure -n ${testName}`);
+  });
+
+  const unsecureServiceName = 'service-unsecure';
+
+  it('Unsecured route should prefix with http', async() => {
```
[the remark this answers is not in the document]
Comment (yapei) [reply frontend/integration-tests/tests/networking.scenario.ts]: Yes, I agree with you! I will close the PR

File: frontend/integration-tests/views/networking.view.ts [hunk 3]
```diff
@@ -0,0 +1,37 @@
+import { $, $$, browser, ExpectedConditions as until } from 'protractor';
```
[the remark this answers is not in the document]
Comment (yapei) [reply frontend/integration-tests/views/networking.view.ts]: will remove `networking.view.ts`

File: frontend/integration-tests/tests/networking.scenario.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (yapei) [reply frontend/integration-tests/tests/networking.scenario.ts]: Will remove `networking.scenario.ts`

File: frontend/integration-tests/tests/networking.scenario.ts [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+    const unsecureRouteName = 'my-unsecure-route';
+    await networkingView.createUnsecureRoute(unsecureRouteName, unsecureServiceName);
+    await networkingView.visitRoutesDetailsPage(appHost, testName, unsecureRouteName);
+    await crudView.isLoaded();
+    const hostSearchLabel = 'Location';
+    const hostnameFoundIndex = await utilsView.getKeyIndex(await $$('dt'), hostSearchLabel, async(label) => {
```
[the remark this answers is not in the document]
Comment (yapei) [reply frontend/integration-tests/tests/networking.scenario.ts]: Will remove `netowrking.scenario.ts`

File: frontend/integration-tests/views/utils.view.ts [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+export const getKeyIndex = async function(iterable, needle, callback) {
+  const array = [...iterable];
+  for (let index = 0; index < array.length; index++) {
+    if (await callback(array[index], index, array)) {
+      return index;
+    }
+  }
+};
+
+export const getResourceJSON = (name: string, namespace: string, kind: string) => {
```
[the remark this answers is not in the document]
Comment (yapei) [reply frontend/integration-tests/views/utils.view.ts]: Updated

File: frontend/integration-tests/views/utils.view.ts [hunk 6]
```diff
@@ -0,0 +1,27 @@
+import { $, browser, by, ExpectedConditions as until, element } from 'protractor';
+import { execSync } from 'child_process';
+import * as _ from 'lodash';
+
+export const chooseFromList = async(itemName: string) => {
+  await browser.wait(until.presenceOf($('li')), 5000);
```
[the remark this answers is not in the document]
Comment (yapei) [reply frontend/integration-tests/views/utils.view.ts]: updated using existing code, WDYT?

[the remark this answers is not in the document]
Comment (yapei) [reply frontend/integration-tests/views/utils.view.ts]: This PR only move some common code to `utils.view.ts`, no other scenarios added

**Pull request closed without merging**